src/core_modules/capture-core/src/components/QuickSelector/OrgUnitSelector.component.js

```diff
@@ -9,56 +11,130 @@ import { FormControl } from 'material-ui-next/Form';
-    selectedOrgUint: Object,
+    handleClickOrgUnit: (orgUnitId: string, orgUnitObject: Object) => void,
+    selectedOrgUint: string,
+    storedOrgUnits: Object,
     classes: Object,
 };
 
 class OrgUnitSelector extends Component<Props> {
-    handleChange: (event: any) => void;
-    handleClick: (orgunit: Object) => void;
+    handleClick: (orgUnit: Object) => void;
+    handleReset: () => void;
     constructor(props) {
         super(props);
-        this.handleChange = this.handleChange.bind(this);
+
+        this.state = {
+			roots: [],
+		};
+
         this.handleClick = this.handleClick.bind(this);
+        this.handleReset = this.handleReset.bind(this);
+
+        const currentUser = getCurrentUser();
+        // Get orgUnits assigned to currentUser and set them as roots to be used by orgUnitTree.
+        currentUser.getOrganisationUnits({
+            fields: 'id,path,displayName,children::isNotEmpty',
+        })
+        .then(roots => roots.toArray())
+        .then(roots => this.setState({
+            roots
+        }));
     }
 
-    handleChange(event) {
-        this.props.handleChangeOrgUnit(event.target.value);
+    handleClick(event, selectedOu) {
+        const orgUnitObject = {id: selectedOu.id, name: selectedOu.displayName, path: selectedOu.path};
+        this.props.handleClickOrgUnit(selectedOu.id, orgUnitObject);
     }
 
-    handleClick(orgunit) {
-        this.props.handleClickOrgUnit(orgunit.id);
+    handleReset() {
+        this.props.handleClickOrgUnit(undefined);
     }
 
     // TODO:Add OrgUnitTree when there are a lot of orgunits
     render() {
         // TODO: Add real data and remove placeholder data.
         const orgUnits = [{ id: 1, name: 'OrgUnit 1' }, { id: 2, name: 'OrgUnit 2' }, { id: 3, name: 'OrgUnit 3' },
             { id: 4, name: 'OrgUnit 4' }, { id: 5, name: 'OrgUnit 5' }, { id: 6, name: 'OrgUnit 6' }];
-        // If less than or equal, display as list.
+
+        // If orgUnit is set in Redux state.
+        if (this.props.selectedOrgUint) {
+            return (
+                <div>
+                    <Paper elevation={1} className={this.props.classes.selectedPaper}>
+                        <h4 className={this.props.classes.title}>{ getTranslation('selected_orgunit') }</h4>
+                        <p className={this.props.classes.selectedText}>{this.props.storedOrgUnits[this.props.selectedOrgUint].name}
+                            <IconButton className={this.props.classes.selectedButton} onClick={() => this.handleReset()}>
+                                <ClearIcon className={this.props.classes.selectedButtonIcon} />
+                            </IconButton>
+                        </p>
+                    </Paper>
+                </div>
+            );
+        }
+        // TODO: Find a way to know how many total orgunits the user is assigned to.
+        // If less than or equal to 5 orgUnits, display as list.
         if (orgUnits.length <= 5) {
```

Comment:
Comment out for now. We need a way of finding all eligable org units, and we have to design what an eligable org unit would be.

Reply:
Done.